pkg/server/api/svid/v1/service.go

```diff
@@ -324,6 +347,64 @@ func (s *Service) NewJWTSVID(ctx context.Context, req *svid.NewJWTSVIDRequest) (
 	}, nil
 }
 
-func (s *Service) NewDownstreamX509CA(context.Context, *svid.NewDownstreamX509CARequest) (*svid.NewDownstreamX509CAResponse, error) {
-	return nil, status.Error(codes.Unimplemented, "not implemented")
+func (s *Service) NewDownstreamX509CA(ctx context.Context, req *svid.NewDownstreamX509CARequest) (*svid.NewDownstreamX509CAResponse, error) {
+	log := rpccontext.Logger(ctx)
+
+	if err := rpccontext.RateLimit(ctx, 1); err != nil {
+		log.WithError(err).Error("Rejecting request due to downstream CA signing rate limit")
+		return nil, err
+	}
+
+	downstreamEntries, isDownstream := rpccontext.CallerDownstreamEntries(ctx)
+	if !isDownstream || downstreamEntries == nil || len(downstreamEntries) == 0 {
+		// The real CallerDownstreamEntries will never return empty, but it helps for writing unit tests
+		err := status.Error(codes.NotFound, "error not found or not authorized")
+		log.WithError(err).Error("Rejecting NewDownstreamX509CA request because caller has no downstream entries")
+		return nil, err
+	}
+
+	entry := downstreamEntries[0]
+
+	spiffeID, err := api.IDFromProto(entry.SpiffeId)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "malformed entry")
+	}
+
+	csr, err := s.parseAndCheckCSR(ctx, req.Csr)
+	if err != nil {
+		return nil, err
+	}
+
+	x509CASvid, err := s.ca.SignX509CASVID(ctx, ca.X509CASVIDParams{
+		SpiffeID:  spiffeID.String(),
```

Comment:
The SPIFFEID does not come from the request or entry and should be determined by the trust domain:
```suggestion
		SpiffeID:  s.td.IDString(),
```